index.js

```diff
@@ -19,23 +19,77 @@ const createRvClient = (
     );
   }
 
-  const sha1 = async (seed) => {
-    const enc = new TextEncoder();
-    const hash = await crypto.subtle.digest('SHA-1', enc.encode(seed));
-    return Array.from(new Uint8Array(hash)).
-        map(v => v.toString(16).padStart(2, '0')).
-        join('');
-  };
+  const pemToArrayBuffer = pem => {
+    // Remove the PEM header and footer
+    const b = pem.replace(/-----(BEGIN|END) [A-Z ]+-----/g, '').replace(/\s+/g, '');
+    const binaryString = window.atob(b);
+    const len = binaryString.length;
+    const bytes = new Uint8Array(len);
+    for (let i = 0; i < len; i++) {
+        bytes[i] = binaryString.charCodeAt(i);
+    }
+    return bytes.buffer;
+  }
+
+  const importPrivateKey =  async (privateKeyBuffer) => {
+    const key = await window.crypto.subtle.importKey(
```

Comment:
What if window is not defined, eg in a nodejs environment? you could do:

```js

const _crypto = typeof window !== 'undefined' ? window.crypto : crypto;

```

Reply:
okay, i thought this needed to be run only on browser

Reply:
No, that's why we will also need to look at packaging (exporting for esm/cjs and browser).

Reply:
As discussed node won't be supported at the moment